services/wallet/datastore.go

```diff
@@ -560,51 +552,26 @@ var (
 	ErrGeoResetDifferent = errors.New("geo reset is different")
 )
 
-// LinkWallet links a wallet together
-func (pg *Postgres) LinkWallet(ctx context.Context, ID string, userDepositDestination string, providerLinkingID uuid.UUID, anonymousAddress *uuid.UUID, depositProvider, country string) error {
-	sublogger := logger(ctx).With().Str("wallet_id", ID).Logger()
-	sublogger.Debug().Msg("linking wallet")
-
-	// rep check
-	if repClient, ok := ctx.Value(appctx.ReputationClientCTXKey).(reputation.Client); ok {
-		walletID, err := uuid.FromString(ID)
-		if err != nil {
-			sublogger.Warn().Err(err).Msg("invalid wallet id")
-			return fmt.Errorf("invalid wallet id, not uuid: %w", err)
-		}
-		// we have a client, check the value for ID
-		reputable, cohorts, err := repClient.IsLinkingReputable(ctx, walletID, country)
-		if err != nil {
-			sublogger.Warn().Err(err).Msg("failed to check reputation")
-			return fmt.Errorf("failed to check wallet rep: %w", err)
-		}
+// LinkWallet links a rewards wallet to the given deposit provider.
+func (pg *Postgres) LinkWallet(ctx context.Context, id string, userDepositDestination string, providerLinkingID uuid.UUID, depositProvider, country string) error {
+	walletID, err := uuid.FromString(id)
+	if err != nil {
+		return fmt.Errorf("invalid wallet id, not uuid: %w", err)
+	}
 
-		var (
-			isTooYoung        = false
-			geoResetDifferent = false
-		)
-		for _, v := range cohorts {
-			if isTooYoung = (v == reputation.CohortTooYoung); isTooYoung {
-				break
-			}
-			if geoResetDifferent = (v == reputation.CohortGeoResetDifferent); geoResetDifferent {
-				break
-			}
-		}
+	repClient, ok := ctx.Value(appctx.ReputationClientCTXKey).(reputation.Client)
+	if !ok {
+		return ErrNoReputationClient
+	}
 
-		if !reputable && !isTooYoung && !geoResetDifferent {
-			sublogger.Info().Msg("wallet linking attempt failed - unusual activity")
-			countLinkingFlaggedUnusual.Inc()
-			return ErrUnusualActivity
-		} else if geoResetDifferent {
-			sublogger.Info().Msg("wallet linking attempt failed - geo reset is different")
-			return ErrGeoResetDifferent
-		}
+	// TODO(clD11): We no longer need to act on the response and only require a successful call to reputation to
+	//  continue linking. As part of the wallet refactor we should clean this up.
+	if _, _, err := repClient.IsLinkingReputable(ctx, walletID, country); err != nil {
```

Comment:
can this be done in the service level now instead of here?  are we losing metrics with this change as to why the failure happened?

Reply:
Yes, the plan is to move a variant of this call into the service layer as part of the refactor. We should also review what we actually need to reputation to do as that endpoint does a lot and with this requirement we might not need all of the functionality. The cohort metrics are recorded in reputation so we should still have the data.